integrations/google_ai/src/haystack_integrations/components/generators/google_ai/chat/gemini.py

```diff
@@ -214,81 +241,29 @@ def from_dict(cls, data: Dict[str, Any]) -> "GoogleAIGeminiChatGenerator":
-            converted_part = Part()
-            converted_part.inline_data.data = part.data
-            converted_part.inline_data.mime_type = part.mime_type
-            return converted_part
-        elif isinstance(part, Part):
-            return part
-        else:
-            msg = f"Unsupported type {type(part)} for part {part}"
-            raise ValueError(msg)
-
-    def _message_to_part(self, message: ChatMessage) -> Part:
-        if message.is_from(ChatRole.ASSISTANT) and message.name:
-            p = Part()
-            p.function_call.name = message.name
-            p.function_call.args = {}
-            for k, v in json.loads(message.text).items():
-                p.function_call.args[k] = v
-            return p
-        elif message.is_from(ChatRole.SYSTEM) or message.is_from(ChatRole.ASSISTANT):
-            p = Part()
-            p.text = message.text
-            return p
-        elif "FUNCTION" in ChatRole._member_names_ and message.is_from(ChatRole.FUNCTION):
-            p = Part()
-            p.function_response.name = message.name
-            p.function_response.response = message.text
-            return p
-        elif message.is_from(ChatRole.TOOL):
-            p = Part()
-            p.function_response.name = message.tool_call_result.origin.tool_name
-            p.function_response.response = message.tool_call_result.result
-            return p
-        elif message.is_from(ChatRole.USER):
-            return self._convert_part(message.text)
-
-    def _message_to_content(self, message: ChatMessage) -> Content:
-        if message.is_from(ChatRole.ASSISTANT) and message.name:
-            part = Part()
-            part.function_call.name = message.name
-            part.function_call.args = {}
-            for k, v in json.loads(message.text).items():
-                part.function_call.args[k] = v
-        elif message.is_from(ChatRole.SYSTEM) or message.is_from(ChatRole.ASSISTANT):
-            part = Part()
-            part.text = message.text
-        elif "FUNCTION" in ChatRole._member_names_ and message.is_from(ChatRole.FUNCTION):
-            part = Part()
-            part.function_response.name = message.name
-            part.function_response.response = message.text
-        elif message.is_from(ChatRole.USER):
-            part = self._convert_part(message.text)
-        elif message.is_from(ChatRole.TOOL):
-            part = Part()
-            part.function_response.name = message.tool_call_result.origin.tool_name
-            part.function_response.response = message.tool_call_result.result
-        else:
-            msg = f"Unsupported message role {message.role}"
-            raise ValueError(msg)
-
-        role = "user"
-        if message.is_from(ChatRole.ASSISTANT) or message.is_from(ChatRole.SYSTEM):
-            role = "model"
-        return Content(parts=[part], role=role)
+    @staticmethod
+    def _convert_tool_to_google_tool(tool: Tool) -> FunctionDeclaration:
```

Comment:
Something I'm wondering myself is this method name too expressive. Don't change it but I remember Silvano's recommendations not to double down on something that is obvious, so _convert_to_google_tool? Some others function ☝️ methods as well. Not a request just speaking out loud.

Reply:
modified this method name accordingly.